File: app/backend/src/main/java/com/app/gamereview/model/Character.java
package com.app.gamereview.model;

import com.app.gamereview.model.common.BaseModel;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;

import java.util.List;
import java.util.Map;

@Document(collection = "Character")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Character extends BaseModel {
    private String name;

    private String icon;

    private String description;

    private List<String> games;

    private String type;

    private String gender;

    private String race;

    private String status;

    private String occupation;

    private String birthDate;

    private String voiceActor;

    private String height;

    private String age;

    @Field("customFields")
    private Map<String, String> customFields;
}

File: app/frontend/src/Services/character.ts
import axios from "axios";

export const createCharacter = async ({
  name,
  type,
  gender,
  race,
  status,
  occupation,
  birthDate,
  voiceActor,
  height,
  age,
  games,
  icon,
  description,
}: {
  name: string;
  type: string;
  gender: string;
  race: string;
  status: string;
  occupation: string;
  birthDate: Date;
  voiceActor: string;
  height: string;
  age: string;
  games: any;
  icon: string;
  description: string;
}) => {
  const response = await axios.post(
    `${import.meta.env.VITE_APP_API_URL}/character/create`,
    {
      name,
      type,
      gender,
      race,
      status,
      occupation,
      birthDate,
      voiceActor,
      height,
      age,
      games,
      icon,
      description,
    }
  );

  return response.data;
};
export async function getCharacterByGame(gameId: string) {
  const response = await axios.get(
    `${import.meta.env.VITE_APP_API_URL}/character/get-game-characters`,
    { params: { gameId: gameId } }
  );
  return response.data;
}
